src/lib/y2partitioner/dialogs/lvm_lv_info.rb

```diff
@@ -0,0 +1,188 @@
+        #   a controller collecting data for a LV to be created
+        def initialize(controller)
+          super()
+          textdomain "storage"
+          @controller = controller
+        end
+
+        # @macro seeAbstractWidget
+        def label
+          _("Logical Volume")
+        end
+
+        # @macro seeAbstractWidget
+        def init
+          self.value = @controller.lv_name
+        end
+
+        # @macro seeAbstractWidget
+        def store
+          @controller.lv_name = value
+        end
+
+        # @macro seeAbstractWidget
+        def validate
+          error_message = nil
+
+          if value.nil? || value.empty?
+            error_message = _("Enter a name for the logical volume.")
+          end
+
+          error_message ||= @controller.error_for_lv_name(value)
+
+          if !error_message && @controller.lv_name_in_use?(value)
+            error_message =
+              _(
+                "A logical volume named \"%{lv_name}\" already exists\n" \
+                "in volume group \"%{vg_name}\"."
+              ) % { lv_name: value, vg_name: @controller.vg_name }
+          end
+
+          if error_message
+            Yast::Popup.Error(error_message)
+            Yast::UI.SetFocus(Id(widget_id))
+            false
+          else
+            true
+          end
+        end
+      end
+
+      # Choose the type of the new logical volume (normal, thin or thin pool)
+      # @note When the selected type is :thin, it also allows to select the
+      #   used thin pool (see {ThinPoolSelector}).
+      class TypeWidget < Widgets::ControllerRadioButtons
+        # @param controller [Sequences::Controllers::LvmLv]
+        #   a controller collecting data for a LV to be created
+        def initialize(controller)
+          textdomain "storage"
+          @controller = controller
+        end
+
+        # @macro seeAbstractWidget
+        def label
+          _("Type")
+        end
+
+        # @see Widgets::ControllerRadioButtons
+        def items
+          [
+            [:normal, _("Normal Volume")],
+            [:thin_pool, _("Thin Pool")],
+            [:thin, _("Thin Volume")]
+          ]
+        end
+
+        # @see Widgets::ControllerRadioButtons
+        def widgets
+          @widgets ||= [
+            CWM::Empty.new("normal_widget"),
+            CWM::Empty.new("thin_pool_widget"),
+            ThinPoolSelector.new(@controller)
+          ]
+        end
+
+        # @macro seeAbstractWidget
+        def init
+          self.value = (@controller.type_choice ||= :normal)
+          # trigger disabling the other subwidgets
+          handle("ID" => value)
+        end
+
+        # @macro seeAbstractWidget
+        def store
+          @controller.type_choice = value
+          @controller.thin_pool = current_widget.value if value == :thin
+        end
+      end
+
+      # Selector for used thin pool
+      # TODO: thin provisioning should be implemented in libstorage-ng
```

Comment:
Well, probably libstorage-ng is already ready to work with thin provisioning, so the comment should be only "TODO" because the implementation of thin pools is out of scope in this PBI.